File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
  public:

  int count=0;
  int maxCount=0;
  int maxDepth(TreeNode* root)
  {
    if (!root) return count;
    count++;
    maxCount=max(count,maxCount);
    maxDepth(root->left);
    maxDepth(root->right);
    count--;


    return maxCount;
  }
};

File: Data Structures & Algorithms/max-water-container/submission-0.cs
public class Solution {
    public int MaxArea(int[] height) {
        int area=0;
int max = 0;

int left = 0;
int right = height.Length - 1;

int h;
int w;
while (left<right)
{

    h = Math.Min(height[left], height[right]);
    w = right - left;

    area = h * w;
    max = Math.Max(max,area);

    if (height[right] >= height[left])
        left++;

    else 
        right--;
}
return max;
    }
}
